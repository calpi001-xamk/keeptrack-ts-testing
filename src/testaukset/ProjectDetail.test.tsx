import { render, screen } from '@testing-library/react';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import ProjectDetail from '../projects/ProjectDetail';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';
import { store } from '../state';
import { MOCK_PROJECTS } from '../projects/MockProjects';
import { Provider } from 'react-redux';


describe('<ProjectDetail/>', () => {
    const setup = () =>
        render(
            <Provider store={store}>
                <MemoryRouter>
                    <ProjectDetail project={MOCK_PROJECTS[0]} />
                </MemoryRouter>
            </Provider>
        );

    beforeEach(() => {
        
    });

    test('Tulisi renderöityä ilman kaatumista', () => {
        setup();
        expect(screen).toBeDefined();
    });

    test('Näkyykö projektin yksittäisiä tietoja', () => {
        setup();
        expect(screen.getByRole('heading')).toHaveTextContent(MOCK_PROJECTS[0].name)
        expect(screen.getByRole('img')).toBeDefined()
        expect(screen.getByText(MOCK_PROJECTS[0].name))
        expect(screen.getByText(MOCK_PROJECTS[0].description))
        expect(screen.getByText(/signed/i)).toBeInTheDocument();
        expect(screen.getByText(/budget/i)).toBeInTheDocument();
    });

    test('snapshot', () => {
        const tree = renderer.create(<ProjectDetail project={MOCK_PROJECTS[0]}/>).toJSON();
        expect(tree).toMatchSnapshot();
      });

})