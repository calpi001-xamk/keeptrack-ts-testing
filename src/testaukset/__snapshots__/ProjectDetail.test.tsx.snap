// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProjectDetail/> snapshot 1`] = `
<div
  className="row"
>
  <div
    className="col-sm-6"
  >
    <div
      className="card large"
    >
      <img
        alt="Johnson - Kutch"
        className="rounded"
        src="/assets/placeimg_500_300_arch4.jpg"
      />
      <section
        className="section dark"
      >
        <h3
          className="strong"
        >
          <strong>
            Johnson - Kutch
          </strong>
        </h3>
        <p>
          Fully-configurable intermediate framework. Ullam occaecati libero laudantium nihil voluptas omnis.
        </p>
        <p>
          Budget : 
          54637
        </p>
        <p>
          Signed: 
          5.8.2013
        </p>
        <p>
          <mark
            className="active"
          >
             
            inactive
          </mark>
        </p>
      </section>
    </div>
  </div>
</div>
`;
